### PR TITLE
Invalidates token when user changes their password

### PR DESCRIPTION
**Pull request by artstorm**

### CODE REVIEW

File: api/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -177,6 +177,12 @@ public function patchOne(Request $request, $id)
         // Extract the credentials and update if necessary
         $credentialUpdateDetails = $request->get('_user_credential', []);
         if (!empty($credentialUpdateDetails)) {
+            // Invalidate token for the user when user changes their password
+            if ($this->jwtAuth->user()->user_id == $model->user_id) {
```
Comment (zakhenry) [inline api/app/Http/Controllers/UserController.php:181]: Is there a case where this wouldn't match?

**Comment by artstorm** [comment]
@xiphiaz This approach invalidates the token when the user changes their password. 

If an admin would change the password it would not be invalidated as we don't have the token in that case. It would probably be a rare case to change another user's password, though not unthinkable? If it is desirable to invalidate a token under that circumstance, we can solve that by adding our own blacklist cache that keeps users that are forced to login on their next request.

File: api/app/Http/Controllers/UserController.php (hunk 1 above)
Comment (zakhenry) [inline api/app/Http/Controllers/UserController.php:181]: Ah actually I get it now, it's for when an admin changes a users password right?

Comment (artstorm) [inline api/app/Http/Controllers/UserController.php:181]: @xiphiaz Yes, if an admin updates the user's password.

**Comment by zakhenry** [comment]
@artstorm no I think the approach you've done here is sufficient. An admin changing another user's password would be ok to have the user continue to use their old token until instructed to log in again.

**Pull request merged**